contracts/core/RebalancingSet.sol

```diff
@@ -0,0 +1,389 @@
+
+        // Set current set to be rebalancing set
+        currentSet = rebalancingSet;
+
+        //Calculate new shareRatio
+        //shareRatio = rebalanceSetSupply.mul(10**18).div(totalSupply_);
+
+        // Update state parameters
+        lastRebalanceTimestamp = block.timestamp;
+        rebalanceState = State.Default;
+
+        // emit RebalanceFinished event
+        emit RebalanceFinished(currentSet, shareRatio);
+    }
+
+    /*
+     * Mint set token for given address.
+     * Can only be called by authorized contracts.
+     *
+     * @param  _issuer      The address of the issuing account
+     * @param  _quantity    The number of sets to attribute to issuer
+     */
+    function mint(
+        address _issuer,
+        uint256 _quantity
+    )
+        external
+    {
+        // Check that function caller is Core
+        require(msg.sender == ISetFactory(factory).core());
+
+        // Check that set is not in Rebalancing State
+        require(rebalanceState != State.Rebalance);
+
+        // Update token balance of the issuer
+        balances[_issuer] = balances[_issuer].add(_quantity);
+
+        // Update the total supply of the set token
+        totalSupply_ = totalSupply_.add(_quantity);
+
+        // Emit a transfer log with from address being 0 to indicate mint
+        emit Transfer(address(0), _issuer, _quantity);
+    }
+
+    /*
+     * Burn set token for given address.
+     * Can only be called by authorized contracts.
+     *
+     * @param  _from        The address of the redeeming account
+     * @param  _quantity    The number of sets to burn from redeemer
+     */
+    function burn(
+        address _from,
+        uint256 _quantity
+    )
+        external
+    {
+        // Check that function caller is Core
+        require(msg.sender == ISetFactory(factory).core());
+
+        // Check that set is not in Rebalancing State
+        require(rebalanceState != State.Rebalance);
+
+        // Require user has tokens to burn
+        require(balances[_from] >= _quantity);
+
+        // Update token balance of user
+        balances[_from] = balances[_from].sub(_quantity);
+
+        // Update total supply of Set Token
+        totalSupply_ = totalSupply_.sub(_quantity);
+
+        // Emit a transfer log with to address being 0 indicating burn
+        emit Transfer(_from, address(0), _quantity);
+    }
+
+    /*
+     * Set new manager address
+     *
+     * @param  _newManager       The address of the redeeming account
+     */
+    function setManager(
+        address _newManager
+    )
+        external
+    {
+        require(msg.sender == manager);
+
+        emit NewManagerAdded(_newManager, manager);
+        manager = _newManager;
+    }
+
+    /*
+     * Get addresses of setToken underlying the Rebalancing Set
+     *
+     * @return  componentAddresses       Array of currentSet
+     */
+    function getComponents()
+        external
+        view
+        returns(address[1])
```

Comment:
Is this more efficient? I think the normal getComponents interface is just address[]. Wonder if there's any issues with address[1]

Reply:
It wouldn't let me return a dynamic address for whatever reason. Perhaps because what's being returned isn't dynamic in nature?

Reply:
This might be an issue when we go to issue (lol)...I'll have to consider ways to be able to return a dynamic address array